util/db/repository.go

```diff
@@ -58,22 +58,12 @@ func (db *db) CreateRepository(ctx context.Context, r *appsv1.Repository) (*apps
 		return nil, status.Errorf(codes.AlreadyExists, "repository '%s' already exists", r.Repo)
 	}
 
-	data := make(map[string][]byte)
-	if r.Username != "" {
-		data[username] = []byte(r.Username)
-	}
-	if r.Password != "" {
-		data[password] = []byte(r.Password)
-	}
-	if r.SSHPrivateKey != "" {
-		data[sshPrivateKey] = []byte(r.SSHPrivateKey)
-	}
-
 	repoInfo := settings.RepoCredentials{
 		URL:                   r.Repo,
 		InsecureIgnoreHostKey: r.IsInsecure(),
 		Insecure:              r.IsInsecure(),
 		EnableLFS:             r.EnableLFS,
+		FetchRefspecs:         append([]string(nil), r.FetchRefspecs...),
```

Comment:
this syntax looks odd, why not just `FetchRefspecs:         r.FetchRefspecs,`

Reply:
Because paranoia. I wanted to copy the input slice by value. Seemed safer. You think this is too much?

Reply:
I think you'll be find with the value

Reply:
Removed all the slice cloning. I really was under the impression that slices are copied by reference?